SwiftLeeds/Views/Components/CommonTileView.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  CommonTileView.swift
+//  SwiftLeeds
+//
+//  Created by Alex Logan on 05/07/2022.
+//
+
+import SwiftUI
+
+/// Generic primary secondary view
+struct CommonTileView<BackgroundType: ShapeStyle>: View {
+    @Environment(\.sizeCategory) var sizeCategory
+
+    let primaryText: String
+    let secondaryText: String?
+    let primaryColor: Color
+    let secondaryColor: Color
+    var backgroundStyle: BackgroundType
+
+    var accessibilityTextEnabled: Bool {
+        sizeCategory >= .accessibilityMedium
+    }
+
+    init(
+        primaryText: String,
+        secondaryText: String?,
+        primaryColor: Color = Color.primary,
+        secondaryColor: Color = Color.secondary,
+        backgroundStyle: BackgroundType
+    ) {
+        self.primaryText = primaryText
+        self.secondaryText = secondaryText
+        self.primaryColor = primaryColor
+        self.secondaryColor = secondaryColor
+        self.backgroundStyle = backgroundStyle
+    }
+
+    init(
+        primaryText: String,
+        secondaryText: String?,
+        primaryColor: Color = Color.primary,
+        secondaryColor: Color = Color.secondary,
+        backgroundStyle: Color = Color.cellBackground
+    ) where BackgroundType == Color {
+        self.primaryText = primaryText
+        self.secondaryText = secondaryText
+        self.primaryColor = primaryColor
+        self.secondaryColor = secondaryColor
+        self.backgroundStyle = backgroundStyle
+    }
+
+    var body: some View {
+        sizeAwareStack(content: {
+            Text(primaryText)
+                .font(.subheadline.weight(.semibold))
+                .foregroundColor(primaryColor)
+            if !accessibilityTextEnabled {
+                Spacer()
+            }
+            if let secondaryText = secondaryText {
+                Text(secondaryText)
+                    .font(.subheadline.weight(.medium))
+                    .foregroundColor(secondaryColor)
+            }
+        })
+        .padding(Padding.cell)
+        .frame(minHeight: Constants.compactCellMinimumHeight)
+        .background(
+            backgroundStyle,
+            in: RoundedRectangle(cornerRadius: Constants.cellRadius)
+        )
+        .accessibilityElement(children: .ignore)
+        .accessibilityLabel(
```

Comment:
There's arguments for `
.accessibilityElement(children: .combine)` here but I prefer the manual control.